rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/VersionedModelControllerV3.scala

```diff
@@ -0,0 +1,218 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.controllers.v3
+
+import com.mongodb.client.result.UpdateResult
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.bind.annotation._
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import za.co.absa.enceladus.model.menas.audit._
+import za.co.absa.enceladus.model.versionedModel._
+import za.co.absa.enceladus.model.{ExportableObject, UsedIn, Validation}
+import za.co.absa.enceladus.rest_api.controllers.BaseController
+import za.co.absa.enceladus.rest_api.controllers.v3.VersionedModelControllerV3.LatestVersionKey
+import za.co.absa.enceladus.rest_api.services.VersionedModelService
+
+import java.net.URI
+import java.util
+import java.util.Optional
+import java.util.concurrent.CompletableFuture
+import javax.servlet.http.HttpServletRequest
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+abstract class VersionedModelControllerV3[C <: VersionedModel with Product
+  with Auditable[C]](versionedModelService: VersionedModelService[C]) extends BaseController {
+
+  import za.co.absa.enceladus.rest_api.utils.implicits._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  // todo maybe offset/limit?
+  @GetMapping(Array(""))
+  @ResponseStatus(HttpStatus.OK)
+  def getList(@RequestParam searchQuery: Optional[String]): CompletableFuture[Seq[VersionedSummary]] = {
+    versionedModelService.getLatestVersionsSummarySearch(searchQuery.toScalaOption)
+  }
+
+  @GetMapping(Array("/{name}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getVersionSummaryForEntity(@PathVariable name: String): CompletableFuture[VersionedSummary] = {
+    versionedModelService.getLatestVersionSummary(name) map {
+      case Some(entity) => entity
+      case None => throw notFound()
+    }
+  }
+
+  @GetMapping(Array("/{name}/{version}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getVersionDetail(@PathVariable name: String,
+                       @PathVariable version: String): CompletableFuture[C] = {
+    forVersionExpression(name, version)(versionedModelService.getVersion).map {
+      case Some(entity) => entity
+      case None => throw notFound()
+    }
+  }
+
+
+  @GetMapping(Array("/{name}/audit-trail"))
+  @ResponseStatus(HttpStatus.OK)
+  def getAuditTrail(@PathVariable name: String): CompletableFuture[AuditTrail] = {
+    versionedModelService.getAuditTrail(name)
+  }
+
+  @GetMapping(Array("/{name}/{version}/used-in"))
```

Comment:
Here, I think it makes sense. 
Since the _disable_ is available for all entity types, so should _used-in_. The nice thing about it is that there can be a common approach to entities:
1. making sure that there is anything preventing from disabling (used-in)
2. dealing with linked references returned by used-in if any
3. disabling the entity knowing that it should succeed (without surprises)

It is quite reassuring for the API user to be able to call `used-in` even if currently for a certain type there cannot be any entity linked, it perhaps may be in the future?

All in all, I am a fan of having `used-in`s everywhere (where _disable_ is).